src/components/layouts/Page/index.js

```diff
@@ -56,21 +39,49 @@ const Page = ({
   currentProgress,
   totalProgress,
 }) => {
+  const [pageContentTopPadding, setPageContentTopPadding] = useState(0);
   const { matchesBreakpoint } = useMediaQuery();
+
   const isDesktop =
     (`(min-width: ${Breakpoints.LARGE}px)`,
     matchesBreakpoint(Breakpoints.LARGE));
+  const willUseSmallHeader = !isHome && !isDesktop;
+
+  const headerRef = useCallback((node) => {
+    if (node) {
+      setPageContentTopPadding(node.getBoundingClientRect().top);
+
+      window.addEventListener('resize', () =>
+        setPageContentTopPadding(node.getBoundingClientRect().top),
+      );
+    }
+  }, []);
 
   return (
     <div css={styles.root}>
-      {renderHeader({ currentProgress, isDesktop, isHome, totalProgress })}
-      {renderPageContent({
-        children,
-        hasBackButton,
-        isDesktop,
-        isHome,
-        onBackButtonClick,
-      })}
+      {willUseSmallHeader && (
+        <Header
+          currentProgress={currentProgress}
+          totalProgress={totalProgress}
+        />
+      )}
+
+      {!willUseSmallHeader && (
+        <div css={isDesktop && styles.headerContainerDesktop}>
+          <div css={isDesktop && styles.headerContentDesktop} ref={headerRef}>
```

Comment:
I had to remove the `renderHeader` function and move that logic into `Page` so that I could use this `ref`.